lib/tcpip/network_wrapper.cc

```diff
@@ -23,6 +25,46 @@ constexpr bool UseIPv6 = CHERIOT_RTOS_OPTION_IPv6;
 // IP thread global lock. See comment in FreeRTOS_IP_wrapper.c.
 extern struct FlagLockState ipThreadLockState;
 
+extern "C" int ipFOREVER(void)
+{
+	// We must interrupt the loop if a reset is ongoing (`Restarting`).
+	// However, we must still allow the loop to run during the early stages
+	// of a reset (`IpThreadKicked`).
+	uint32_t state = restartState.load();
+	return (state == 0) || ((state & IpThreadKicked) != 0);
+}
+
+/**
+ * State machine of the restart process. Used for synchronization across the
+ * TCP/IP stack and with the firewall.
```

Comment:
What values can this have?  Is it just a counter?

Reply:
True, it's technically just a counter because the bits should be sequentially set. First `NotRestarting`, then `Restarting`, then `Restarting | IpThreadKicked`, then `Restarting | IpThreadKicked | DriverKicked`, then `NotRestarting` again. Still, I quite like the bit enum as it is. Is there a reason to go for the counter instead that I'm not seeing?

One thing which can be done for sure is reducing the size of the atomic for 32 bit to 8 bit.

Reply:
Actually the main reason why I did this is because I just took what we were doing here: https://github.com/CHERIoT-Platform/network-stack/blob/main/lib/tcpip/startup.cc#L40

Although clearly here we don't need the flag stuff because we have a strictly sequential process.